pkg/networkservice/utils/metadata/context.go

```diff
@@ -34,15 +34,6 @@ func store(parent context.Context, id string, mdMap *metaDataMap) context.Contex
 	return parent
 }
 
-func del(parent context.Context, id string, mdMap *metaDataMap) context.Context {
-	_, ok := parent.Value(metaDataKey{}).(*metaData)
-	if !ok {
-		md, _ := mdMap.LoadAndDelete(id)
-		return context.WithValue(parent, metaDataKey{}, md)
-	}
-	return parent
```

Comment:
Its seems to me the better fix here would be:

```suggestion
return context.WithValue(parent,metaDataKey{},&metaData{})
```

Reply:
I don't think you quite understood my suggestion... I believe that *just* this change will solve your panic issue, no need for other changes.  Small, simple, targeted.

Reply:
Oh wait... I read to quickly... you did fix this :)

Reply:
So actually this fix leads to the same error, because on L40 we are getting `md == nil` and so on L41 we just storing it into the context.
But this direction looks better, so:
```go
if _, ok := parent.Value(metaDataKey{}).(*metaData); !ok {
	if md, ok := mdMap.LoadAndDelete(id); ok {
		return context.WithValue(parent, metaDataKey{}, md)
	}
	return context.WithValue(parent, metaDataKey{}, new(metaData))
}
return parent
```